src/python/pants/jvm/target_types.py

```diff
@@ -259,3 +260,41 @@ class DeployJarTarget(Target):
         "The JAR will contain class files for both first-party code and "
         "third-party dependencies, all in a common directory structure."
     )
+
+
+# -----------------------------------------------------------------------------------------------
+# `jvm_war` targets
+# -----------------------------------------------------------------------------------------------
+
+
+class JvmWarDependenciesField(Dependencies):
+    pass
+
+
+class JvmWarDescriptorAddressField(StringField):
+    alias = "descriptor"
+    help = "Address of a `file` target containing the descriptor (i.e., web.xml) for this WAR file."
+
+
+class JvmWarContentField(SpecialCasedDependencies):
+    alias = "content"
+    help = (
+        "A list of addresses to `resource` and `file` targets with content to place in the "
+        "document root of this WAR file."
```

Comment:
Here, is it fine to use `resources` and `files` too? If so, we usually use `resource / resources`.

Reply:
It should be, yes. Will update to use the plural target type names.